src/python/code/gnoll/__init__.py

```diff
@@ -87,36 +85,47 @@ def extract_from_dice_file(lines, seperator):
         v = [list(map(make_native_type, x)) for x in v]
         return v
 
-    die_file = temp.name
-    os.remove(die_file)
-
-    out_file = str(die_file).encode("ascii")
-    if verbose:
-        print("Rolling: ", s)
-        print("Output in:", out_file)
-
-    s = s.encode("ascii")
-
-    return_code = libc.roll_full_options(
-        s,
-        out_file,
-        verbose,  # enable_verbose
-        breakdown,  # enable_introspect
-        mock is not None,  # enable_mock
-        builtins,  # enable_builtins
-        mock,
-        mock_const,
-    )
-    if return_code != 0:
-        raise_gnoll_error(return_code)
-
-    with open(out_file, encoding="utf-8") as f:
-        lines = f.readlines()
-
-    dice_breakdown = extract_from_dice_file(lines, ",")
-    result = extract_from_dice_file(lines, ";")
-
-    return int(return_code), result, dice_breakdown
+    try:
+        temp = tempfile.NamedTemporaryFile(prefix="gnoll_roll_",
+                                           suffix=".die",
+                                           delete=False)
+        temp.close()
+
+        die_file = temp.name
+
+        out_file = str(die_file).encode("ascii")
+        if verbose:
+            print("Rolling: ", s)
+            print("Output in:", out_file)
+
+        s = s.encode("ascii")
+
+        return_code = libc.roll_full_options(
+            s,
+            out_file,
+            verbose,  # enable_verbose
+            breakdown,  # enable_introspect
+            mock is not None,  # enable_mock
+            builtins,  # enable_builtins
+            mock,
+            mock_const,
+        )
+        if return_code != 0:
+            raise_gnoll_error(return_code)
+
+        with open(out_file, encoding="utf-8") as f:
+            lines = f.readlines()
+
+        dice_breakdown = extract_from_dice_file(lines, ",")
+        result = extract_from_dice_file(lines, ";")
+
+        return int(return_code), result, dice_breakdown
+
+    finally:
+        if not keep_temp_file:
+            if verbose:
+                 print("Deleting:", out_file)

```

Comment:
```suggestion
                print("Deleting:", out_file)

```
5 spaces rather than 4